pkg/client/run.go

```diff
@@ -302,3 +307,18 @@ func humanReadableStatus(str string) string {
 		return fmt.Sprintf("Sonobuoy is in unknown state %q. Please report a bug at github.com/vmware-tanzu/sonobuoy", str)
 	}
 }
+
+
+func getPodStatus(pod corev1.Pod) string {
```

Comment:
This isn't yet sufficient to what we need to show the users.

The most common use case is an ErrImagePull state and that isn't reflected here since that is on the container status:

```
    containerStatuses:
    - image: schnake/sonobuoy:987
      imageID: ""
      lastState: {}
      name: kube-sonobuoy
      ready: false
      restartCount: 0
      started: false
      state:
        waiting:
          message: Back-off pulling image "schnake/sonobuoy:987"
          reason: ImagePullBackOff
```

To repro this use case just run:
```
sonobuoy run --wait --sonobuoy-image=schnake/sonobuoy:987
```

And all I see is:
```
11:19:44 Waiting for the aggregator status to become Running. Currently the status is sonobuoy/sonobuoy: Pending: Ready: ContainersNotReady, containers with unready status: [kube-sonobuoy]
...
...
...
```

Reply:
In the above output I'll also say that you can see how its confusing because the "condition" is "Ready" but its false. As printed, it looks like youre saying it IS ready.

When printing the condition you should just tweak the format string so it reads like "status is pending: [Ready=false]: ContainersNotReady...

Reply:
I've pushed a 4th draft, let me know what you think of it.
I've added also details for the containers that are not ready, so it would show `ImagePullBackOff`, for example.
To do that, I decided to add newlines (to separate the pod details from the container details), let me know what you think of that too (can we have newlines?)
It should look something like this (from the test output)

```
Status: Pending, Reason: ContainersNotReady, containers with unready status: [kube-sonobuoy]
Details of containers that are not ready:
kube-sonobuoy: waiting: ImagePullBackOff, Back-off pulling image "schnake/sonobuoy:987"
Status: Pending, Reason: ContainersNotReady, containers with unready status: [kube-sonobuoy]
Details of containers that are not ready:
kube-sonobuoy: waiting: ContainerCreating
Status: Pending, Reason: Unschedulable, 0/1 nodes are available: 1 node(s) had taint {node.kubernetes.io/not-ready: }, that the pod didn't tolerate.
Status: Running
Status: Running
```
The "Status: Running" should Never Happen™, I left it there to have a return for any code path